theopensystemslab/planx-new
# chore: retire BOPS v1 submission endpoints and events

We have the go ahead from BOPS to stop sending parallel application payloads and only submit via their v2 endpoint.

**Changes here:**
- Creates a single Hasura event when sending to BOPS rather than two
  - Also applies to submission events created by Invite to Pay payment requests
- `/send/bops` endpoint is removed, and `/send/bops-v2` is renamed to `/send/bops` for simplicity
- `/admin/session/:sessionId/bops` endpoint is removed (we use existing `/admin/session/:sessionId/digital-planning-application` endpoint to inspect v2 payload)
- Removes tests and mock data that were v1 specific

I'll open a followup PR to additionally clean up planx-core, but this one can merge anytime before that as we're no longer calling `$api.export.bopsPayload(sessionId)` & v1 submissions are already failing on file tag validation.

**Testing:**
- Go through a submission service & confirm a single BOPS event is created on Send
- Repeat using Invite to Pay (you'll have to successfully complete mock payment)
- Confirm that submission succeeds (valid digital planning payload and audit table entry)
